### ISSUE TITLE
Store additional information in the tabulator / have a tabulator.id

### ISSUE THREAD

**Issue by sistason** [olifolkerd/tabulator#532]
I currently use multiple tabulators in my project.
This makes it important to distinguish, for which tabulator a function is clicked, as they hold different objects, differing mostly by the database_id in the backend, and I use the same functions for most of the tabulators (rowFormatters and cellClicked, for example)

Currently, every data object has the database_id, so I can get that via cell.getData() and know which tabulator I am "in". Now I want to combine objects from multiple tabulators in one combined-tabulator, so they share the database_id.

Optimally, I would Love to have a tabulator-setup-option to just pass a "init-object", which I can get with cell.getTableInfo() or smth. Or even give the tabulator an id, accessible via cell.getTableId() or cell.id (not to confuse with cell.getIndex() obv.), so I could lookup tabulator-information in a global tabulator_information-Array via the id of the tabulator.

Do you think you could add such a feature? 
Would it canonically fit into the tabulator or would you suggest a different approach when generating tabulators via `for (i in tables) { $(".tabulator_" + i).tabulator()... }`?

**Comment by olifolkerd** [comment]
Using the component object you can find the parent table of any cell or row from within a callback, for example inside a custom cell click function:
```js
cellClickCallback = function(e, cell){
       var tabulatorElement = cell.getElement().closest(".tabulator");
}
```
Let me know if that helps.

Cheers

Oli

**Comment by sistason** [comment]
This works just the way I want.

I did test it with the tooltips function, but which gets build at creation-time, I think, at which time the .tabulator does not yet exist, so it did take a while to try different functions and see that it works there ;)

This now works, thanks! 👍 
```
var tabulator = cell_row.getElement().closest(".tabulator").get();
if (!tabulator.length)
    return -1;

var $tabulator = tabulator[0];
return all_tabulators.findIndex(function($tab_){
    return $tab_ == $tabulator;
});
```

**Comment by sistason** [comment]
I just noticed this, but the problem wasn't creation-time or something.

```javascript
cell.getElement().closest(".tabulator")
```

The suggested cell.getElement does not actually work with .closest()!
```javascript
console.log(cell.getElement().closest('.tabulator'));
console.log(cell.getRow().getElement().closest('.tabulator'));
```
returns
```javascript
Object { length: 0, prevObject: Object[1] }
Object { 0: <div#directory_tabulator.tabulator>, length: 1, prevObject: Object[1] }
```

-> .closest() actually only works on the row.getElement(), not on the cell.getElement()!

Noted here for future visitors, as this was hard to catch ;)

**Comment by olifolkerd** [comment]
Hey @sistason,

Thanks for the extra info, this is actually happening because in version 3.2 Tabulator moved to storing the cell as a DOM node rather than as a jQuery element as it dramatically improved render time.

This has cause a regression where the node is being returned instead of the jQuery element in the ***getElement*** function. I will ensure this is fixed for the 3.3 release.

In the mean time the following should work for getting a cell's jQuery wrapped element:
```
var cellElement = $(cell.getElement());
```

Cheers

Oli

**Comment by sistason** [comment]
or just using cell.getRow().getElement() :)

and of course you will fix that. Because it's tabulator you fix everything and you're awesome :D

**Comment by olifolkerd** [comment]
This has now been fixed in today's 3.3 release :)
